internal/service/loadbalancer/backend_tls_config.go

```diff
@@ -2,153 +2,261 @@ package loadbalancer
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to unmarshal loadbalancer backend name",
+			utils.ErrorDiagnosticDetail(err),
+		)
+
+		return
 	}
 
-	if diags = setBackendTLSConfigResourceData(d, t); len(diags) > 0 {
-		return diags
+	tlsConfig, err := r.client.GetLoadBalancerBackendTLSConfig(ctx, &request.GetLoadBalancerBackendTLSConfigRequest{
+		ServiceUUID: loadbalancer,
+		BackendName: backendName,
+		Name:        data.Name.ValueString(),
+	})
+	if err != nil {
+		if utils.IsNotFoundError(err) {
+			resp.State.RemoveResource(ctx)
+		} else {
+			resp.Diagnostics.AddError(
+				"Unable to read loadbalancer backend TLS config details",
+				utils.ErrorDiagnosticDetail(err),
+			)
+		}
+		return
 	}
 
-	return diags
+	resp.Diagnostics.Append(setBackendTLSConfigValues(ctx, &data, tlsConfig)...)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
 }
 
-func resourceBackendTLSConfigUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) (diags diag.Diagnostics) {
-	svc := meta.(*service.Service)
-	var serviceID, beName, name string
-	if err := utils.UnmarshalID(d.Id(), &serviceID, &beName, &name); err != nil {
-		return diag.FromErr(err)
+func (r *backendTLSConfigResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data backendTLSConfigModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	var loadbalancer, backendName string
+	err := utils.UnmarshalID(data.Backend.ValueString(), &loadbalancer, &backendName)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to unmarshal loadbalancer backend name",
+			utils.ErrorDiagnosticDetail(err),
+		)
+
+		return
 	}
-	t, err := svc.ModifyLoadBalancerBackendTLSConfig(ctx, &request.ModifyLoadBalancerBackendTLSConfigRequest{
-		ServiceUUID: serviceID,
-		BackendName: beName,
-		Name:        name,
+
+	apiReq := &request.ModifyLoadBalancerBackendTLSConfigRequest{
+		ServiceUUID: loadbalancer,
+		BackendName: backendName,
+		Name:        data.Name.ValueString(),
 		Config: request.LoadBalancerBackendTLSConfig{
-			Name:                  d.Get("name").(string),
-			CertificateBundleUUID: d.Get("certificate_bundle").(string),
+			Name:                  data.Name.ValueString(),
+			CertificateBundleUUID: data.CertificateBundle.ValueString(),
 		},
-	})
-	if err != nil {
-		return diag.FromErr(err)
 	}
 
-	d.SetId(utils.MarshalID(serviceID, beName, t.Name))
-
-	if diags = setBackendTLSConfigResourceData(d, t); len(diags) > 0 {
-		return diags
+	tlsConfig, err := r.client.ModifyLoadBalancerBackendTLSConfig(ctx, apiReq)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to modify loadbalancer backend TLS config",
+			utils.ErrorDiagnosticDetail(err),
+		)
+		return
 	}
 
-	tflog.Info(ctx, "backend TLS config updated", map[string]interface{}{"name": t.Name, "service_uuid": serviceID, "be_name": beName})
-	return diags
+	resp.Diagnostics.Append(setBackendTLSConfigValues(ctx, &data, tlsConfig)...)
```

Comment:
Is the ID now updated after rename like it was done in SDKv2 version?

Reply:
See reply below on frontend tls config, fixed.